README.md

```diff
@@ -118,19 +118,37 @@ The script waits up to 120 secs (4 x 30 seconds) for the database containers to
 TIMEOUT=${TIMEOUT:-30}
 ```
 
-**Update your hosts file:**
+#### 8) Configure name resolution
 
-Add the following to your `/etc/hosts` file:
+For host name resolution of the containers inside the docker network, this setup includes [DPS](http://mageddo.github.io/dns-proxy-server/latest/en/), a DNS Proxy server. All wikis at `*.web.mw.localhost` will automatically resolve to the nginx-proxy container's IP address. The web container's name is `mediawiki.mw.localhost` and all other containers can be reached from each other via their corresponding host names, defined in `docker-compose.yml`:
+ - `dps.mw.localhost`
+ - `db-master.mw.localhost`
+ - `db-slave.mw.localhost`
+ - `phpmyadmin.mw.localhost`
+ - `graphite.mw.localhost`
+ - `proxy.mw.localhost`
+ - `redis.mw.localhost`
+
+In order to access the containers by name from your docker host system, there are two different ways:
 
+##### Option a) Via /etc/hosts
+
+Add the following to your `/etc/hosts` file:
 ```text
 127.0.0.1 default.web.mw.localhost # mediawiki-docker-dev
 127.0.0.1 proxy.mw.localhost # mediawiki-docker-dev
 127.0.0.1 phpmyadmin.mw.localhost # mediawiki-docker-dev
 127.0.0.1 graphite.mw.localhost # mediawiki-docker-dev
 ```
+You can use the `./hosts-sync` script to try and update it automatically if possible. You may need to use `sudo ./hosts-sync` instead, if the file is not writable by the shell user.
 
-You can also use the `./hosts-sync` script to try and update it automatically if possible. You may
-need to use `sudo ./hosts-sync` instead if the file is not writable by the shell user.
+##### Option b) Via DPS
+
+Alternatively, you can use DPS as your system's primary DNS server to resolve the docker container host names. That feature is disabled by default - to enable simply uncomment the following line in `docker-compose.yml`:
+```text
+      # - /etc/resolv.conf:/etc/resolv.conf
```

Comment:
Its probably advised to suggest doing this with a docker-compose.override file rather than touching the docker-compose file that comes in the git repo (to avoid having to merge things etc when pulling updates in the future!

Reply:
done